src/core/thread/meshcop_dataset.cpp

```diff
@@ -121,6 +121,14 @@ void Dataset::Get(otOperationalDataset &aDataset)
             break;
         }
 
+        case Tlv::kChannelMask:
+        {
+            const uint8_t *entry = reinterpret_cast<const uint8_t *>(cur) + sizeof(Tlv) + sizeof(ChannelMaskEntry);
```

Comment:
While the API only support channel page 0, the implementation needs to be able to handle reading a Channel Mask TLV that contains more than one entry.